core/src/net/buffer.rs

```diff
@@ -42,27 +42,27 @@ impl BufferConfig {
         }
     }
 
-    /// Creates a new BufferConfig with specified values and performs validation
-    pub fn new(
-        chunk_size: usize,
-        initial_pool_count: usize,
-        max_pool_count: usize,
-        encode_min_remaining: usize,
-    ) -> Self {
-        let buffer_config = BufferConfig {
-            chunk_size,
-            initial_pool_count,
-            max_pool_count,
-            encode_min_remaining,
-        };
-        buffer_config.validate();
-        buffer_config
+    /// Sets the BufferConfigs chunk_size to the given value
```

Comment:
Explain units in docs (in the others as well I suppose, won't mark it everyonewhere)